influxql/iterator.go

```diff
@@ -415,6 +415,40 @@ func DrainIterator(itr Iterator) {
 	}
 }
 
+// DrainIterators reads all points from all iterators.
+func DrainIterators(itrs []Iterator) {
+	for {
+		var hasData bool
+
+		for _, itr := range itrs {
+			switch itr := itr.(type) {
+			case FloatIterator:
+				if p := itr.Next(); p != nil {
+					hasData = true
+				}
+			case IntegerIterator:
+				if p := itr.Next(); p != nil {
+					hasData = true
+				}
+			case StringIterator:
+				if p := itr.Next(); p != nil {
+					hasData = true
+				}
+			case BooleanIterator:
+				if p := itr.Next(); p != nil {
+					hasData = true
+				}
+			default:
+				panic(fmt.Sprintf("unsupported iterator type for draining: %T", itr))
+			}
+		}
+
+		if !hasData {
```

Comment:
It continues draining as long as any iterator has points remaining. All iterators should output an equal number of points but I put it in just in case. I'll add a comment.